### ISSUE TITLE
repeated alerts on resolved data

### ISSUE THREAD

**Issue by amirmatin** [Apercu/react-native-zeroconf#47]
first, thanks for the great plugin, works really well ..
I am facing a strange issue and it is when I call zero conf scan and on.resolve, I alert the resolved data however the for the number of times I call the method, number of alerts increment. ie the first time I call method i get one alert, second time i call the method i get 2 alerts at the same time and so on...
BTW there is a time gap between my calls (3 secs ..) 

below is the method that I call..

  method_zeroConf(type = "osc", protocol = "tcp", domain = "local.") {
    //zeroconf.on('stop', () => alert('stop any running scan'));
    try {
      zeroconf.stop();
      //zeroconf.removeDeviceListeners();
      zeroconf.scan(type, protocol, domain);
      //zeroconf.on('start', () => alert('started'));
      //zeroconf.on('found', () => alert('found.......'));
      zeroconf.on("resolved", data => alert(JSON.stringify(data, null, "\t")));
    }
    catch(err) {
      alert('z-conf was not successful !', err);
    }
  }  

now is there a way to clear the data before scan ? 
thanks.

**Comment by Apercu** [comment]
Hum that's weird, are you sure you are not creating multiple zeroconf instances somehow?

**Comment by amirmatin** [comment]
Don't think so..., its only created when the page is loading..  before defining my class..

**Comment by nomoreboredom** [comment]
@amirmatin, I solved this by calling 
```  
componentDidMount() {
  this.zeroconf.removeDeviceListeners();
  this.zeroconf.addDeviceListeners();
} 
 ```
and then again calling ```this.zeroconf.removeDeviceListeners(); ``` on resolve function.

**Comment by Apercu** [comment]
This [conditional](https://github.com/Apercu/react-native-zeroconf/blob/develop/src/index.js#L22) should take care of making sure there is only one listener for each event, @nomoreboredom you were getting the issue too?

**Comment by nomoreboredom** [comment]
@Apercu Yes, however I was building the Zeroconf Object and using the resolve function on the constructor, if that makes a difference...

**Comment by amirmatin** [comment]
@Apercu Thanks for your response.. I still have the issue, I added the componentDidMount before my render () and removeDeviceListeners() after resolve() but still get duplicates..
can you please share a sample code ? 

thanks..

**Comment by Moni-h** [comment]
I also have that problem. How I can solve it?
[Fri Apr 16 2021 11:46:35.967]  LOG      [Start]
[Fri Apr 16 2021 11:46:35.971]  LOG      [Start]
[Fri Apr 16 2021 11:46:40.985]  LOG      [Stop]
[Fri Apr 16 2021 11:46:41.100]  LOG      [Stop]

**Comment by Moni-h** [comment]
> This [conditional](https://github.com/Apercu/react-native-zeroconf/blob/develop/src/index.js#L22) should take care of making sure there is only one listener for each event, @nomoreboredom you were getting the  How I can solve it?
[Fri Apr 16 2021 11:46:35.967] LOG [Start]
[Fri Apr 16 2021 11:46:35.971] LOG [Start]
[Fri Apr 16 2021 11:46:40.985] LOG [Stop]
[Fri Apr 16 2021 11:46:41.100] LOG [Stop]


> Hum that's weird, are you sure you are not creating multiple zeroconf instances somehow?

I call zero conf scans and on resolve, I log the resolved data however the for the number of times I call the method.
[Fri Apr 16 2021 11:46:35.967] LOG [Start]
[Fri Apr 16 2021 11:46:35.971] LOG [Start]
[Fri Apr 16 2021 11:46:40.985] LOG [Stop]
[Fri Apr 16 2021 11:46:41.100] LOG [Stop]

**Comment by Moni-h** [comment]
I call zero conf scans and on resolve, I log the resolved data however the for the number of times I call the method.
[Fri Apr 16 2021 11:46:35.967] LOG [Start]
[Fri Apr 16 2021 11:46:35.971] LOG [Start]
[Fri Apr 16 2021 11:46:40.985] LOG [Stop]
[Fri Apr 16 2021 11:46:41.100] LOG [Stop]
